src/services/expenses.js

```diff
@@ -0,0 +1,75 @@
+'use strict';
+
+let expenses = [];
+
+const getFilteredExpenses = (data) => {
+  let filteredExpenses = expenses;
+
+  const { userId, categories, from, to } = data;
+
+  if (userId) {
+    filteredExpenses = filteredExpenses.filter(
+      (expense) => expense.userId === Number(userId)
+    );
+  }
+
+  if (categories) {
+    filteredExpenses = filteredExpenses.filter(
+      (expense) => expense.category === categories
+    );
+  }
+
+  if (from && to) {
+    filteredExpenses = filteredExpenses.filter((expense) => {
+      return expense.spentAt >= from && expense.spentAt <= to;
+    });
+  }
+
+  return filteredExpenses;
+};
+
+const getById = (expenseId) =>
+  expenses.find((expense) => expense.id === Number(expenseId)) || null;
+
+const create = (data) => {
+  const { title, amount, spentAt, category, userId, note } = data;
+
+  const newExpense = {
+    id: expenses.length + 1,
+    title,
+    amount: amount || 0,
+    spentAt: spentAt || new Date(Date.now()).toISOString(),
+    category: category || '',
+    userId: userId || 0,
+    note: note || '',
```

Comment:
you already check it in the upper lvl
```suggestion
    amount,
    spentAt,
    category,
    userId,
    note,
```